src/vectorcode/chunking.py

```
@@ -279,6 +279,40 @@ def __load_file_lines(self, path: str) -> list[str]:
             lines = fin.readlines()
         return lines
 
+
+    def __get_parser_from_config(self, file_path: str):
+        """
+        Get parser based on filetype_map config.
+        """
+        filetype_map = self.config.filetype_map
+        if not filetype_map:
+            logger.debug("filetype_map is empty in config.")
+            return None
+
+        filename = os.path.basename(file_path)
+        extension = os.path.splitext(file_path)[1]
+        if extension.startswith('.'):
+            extension = extension[1:]
+        logger.debug(f"Checking filetype map for extension '{extension}' in {filename}")
+        for _language, patterns in filetype_map.items():
+            language =  _language.lower()
+            for pattern in patterns:
+                try:
+                    if re.search(pattern, extension):
+                        logger.debug(f"'{filename}' extension matches pattern '{pattern}' for language '{language}'. Attempting to load parser.")
+                        parser = get_parser(language)
+                        logger.debug(f"Found parser for language '{language}' from config.")
+                        return parser
+                except re.error as e:
+                    logger.error(f"Invalid regex pattern '{pattern}' for language '{language}' in filetype_map: {e}")
+                    raise ValueError(f"Invalid regex pattern '{pattern}' for language '{language}' in filetype_map: {e}")
```

There's no need to re-raise a different error. Doing so will get you sth like "when handling this error, we got another error:" with redundant tracebacks, which is IMO redundant and may be overwhelming. Simply put a `raise` statement with no operand will re-raise the same error, and [`main.py` will handle the logging](https://github.com/Davidyz/VectorCode/blob/dfff6494d50299162322e5da720bd173e2d993cc/src/vectorcode/main.py#L112). If you want to add a custom message to the error traceback, you can use `e.add_note()`. See [here](https://github.com/Davidyz/VectorCode/blob/dfff6494d50299162322e5da720bd173e2d993cc/src/vectorcode/common.py#L159) as an example.

---

you are right, sorry. I changed that in a new commit below.